clouddriver-kubernetes/src/test/java/com/netflix/spinnaker/clouddriver/kubernetes/artifact/KubernetesVersionedArtifactConverterTest.java

```diff
@@ -16,52 +16,141 @@
+      KubernetesVersionedArtifactConverter.INSTANCE;
+  private static final ObjectMapper mapper = new ObjectMapper();
+
+  private static final String ACCOUNT = "my-account";
+  private static final String NAMESPACE = "ns";
+  private static final String NAME = "name";
+  private static final String KIND = "Pod";
+  private static final String ARTIFACT_TYPE = "kubernetes/pod";
 
   @Test
   public void checkArtifactAcrossAccount() {
-    KubernetesVersionedArtifactConverter converter = KubernetesVersionedArtifactConverter.INSTANCE;
     ArtifactProvider provider = mock(ArtifactProvider.class);
-    List<Artifact> artifactList = new ArrayList<>();
-    String name = "myname";
-
-    artifactList.add(
-        Artifact.builder()
-            .metadata(ImmutableMap.of("account", "account1"))
-            .version("v003")
-            .build());
-    artifactList.add(
-        Artifact.builder()
-            .metadata(ImmutableMap.of("account", "account2"))
-            .version("v005")
-            .build());
-
-    when(provider.getArtifacts(KubernetesV2Provider.PROVIDER_NAME + "/pod", name, "ns"))
-        .thenReturn(artifactList);
-    Map<String, Object> manifestMap =
-        ImmutableMap.of(
-            "kind", "Pod", "metadata", ImmutableMap.of("name", name, "namespace", "ns"));
-    ObjectMapper mapper = new ObjectMapper();
-    KubernetesManifest manifest = mapper.convertValue(manifestMap, KubernetesManifest.class);
+    when(provider.getArtifacts(ARTIFACT_TYPE, NAME, NAMESPACE))
+        .thenReturn(
+            ImmutableList.of(
+                Artifact.builder().putMetadata("account", "account1").version("v003").build(),
+                Artifact.builder().putMetadata("account", "account2").version("v005").build()));
 
-    Artifact artifact = converter.toArtifact(provider, manifest, "account1");
+    Artifact artifact = converter.toArtifact(provider, getStubManifest(), "account1");
     assertThat(artifact).isNotNull();
     assertThat(artifact.getVersion()).isEqualTo("v004");
   }
+
+  @Test
+  void inferVersionedArtifactProperties() {
+    String name =
+        converter.getDeployedName(
+            Artifact.builder().type("kubernetes/replicaSet").name("my-rs").version("v000").build());
+    assertThat(name).isEqualTo("my-rs-v000");
+  }
+
+  @Test
+  void handlesDashesInName() {
+    String name =
+        converter.getDeployedName(
+            Artifact.builder()
+                .type("kubernetes/replicaSet")
+                .name("my-other-rs-_-")
+                .version("v010")
+                .build());
+    assertThat(name).isEqualTo("my-other-rs-_--v010");
+  }
+
+  @Test
+  void findsMatchingVersionByEquality() {
+    KubernetesManifest manifest1 = getStubManifest();
+    KubernetesManifest manifest2 = getStubManifest();
+    manifest1.put("data", ImmutableMap.of("key", 1, "value", 2));
+    manifest2.put("data", ImmutableMap.of("key", 3, "value", 2));
```

Comment:
Are we doing this just so we don't have two otherwise-identical stub manifests?

Reply:
Yeah, this was converted from Groovy.  Ideally it would change something more meaningful in the manifest rather than this random `data` key, but it is indeed to make sure they are different.

Reply:
Pushed a commit to add a comment as it was kind of confusing.  (Also removed the "value: 2" from both which was confusing.)